### PR TITLE
Task/merge external pr 786

### PR DESCRIPTION
**Pull request by stefaniacardenas**
# Problem
When a user overrides _some_ locale strings but not all of them, a warning is displayed in the console notifying the user that they did not override all strings in the respective locale file.

# Solution

Allow clients to override only some strings for supported languages without seeing any warning. If  an unsupported language locale is used, they will still see the warning.

This PR includes the changes from external PR #786 and would be a fix for #532

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [ ] Have new automated tests been implemented?
- [ ] Have new manual tests been written down?
- [ ] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-795-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-795-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://795-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-795-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-795-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://795-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-795-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-795-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://795-pr-onfido-sdk-ui-onfido.surge.sh

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,9 @@ This project adheres to the Node [default version scheme](https://docs.npmjs.com
 
 ## [Next version]
 
+### Changed
+Public: Disable warning for partial string customisation for supported languages. This change disables warning for clients that override only some strings for a supported language. If they provide partial translations for an unsupported language, they will still see the warning.
+
```
Comment (kopijunkie) [inline CHANGELOG.md]: Minor: Seems pretty wordy. Could be shortened to just 
```
Disable console warning for client integrations that override only some strings for a supported language. If they provide partial translations for an unsupported language, warning is still displayed.
```

File: src/locales/polyglot.js [hunk 2]
```diff
@@ -30,19 +30,22 @@ const extendPolyglot = (locale, polyglot, phrases, mobilePhrases) => {
   return polyglot
 }
 
-const findMissingKeys = (defaultKeys, customKeys) => {
+const findMissingKeys = (defaultKeys, customKeys, language) => {
   const newTranslationsSet = new Set(customKeys)
   const missingKeys =  defaultKeys.filter(element => !newTranslationsSet.has(element))
-  if (missingKeys.length) { console.warn('Missing keys:', missingKeys) }
+  const isSupportedLanguage = Object.keys(availableTransations).some(lang => lang === language)
```
Comment (rfreitas) [inline src/locales/polyglot.js]: Minor: I think we should try to distinguishing between `lang` and `language`. maybe rename `lang` to `aSupportedLanguage`

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-795-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-795-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://795-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**